src/gui/sidebar/previews/layer/SidebarPreviewLayers.h

```diff
@@ -82,4 +89,31 @@ class SidebarPreviewLayers: public SidebarPreviewBase, public LayerCtrlListener
     bool stacked;
 
     IconNameHelper iconNameHelper;
+
+
+    /**
+     * The context menu to display when a layer is right-clicked.
+     */
+    GtkWidget* const contextMenu = nullptr;
+
+
+    GtkWidget* contextMenuMoveUp = nullptr;
+    GtkWidget* contextMenuMoveDown = nullptr;
+    GtkWidget* contextMenuMergeDown = nullptr;
+    GtkWidget* contextMenuDuplicate = nullptr;
+    GtkWidget* contextMenuDelete = nullptr;
+
+    /**
+     * The data passed to the menu item callbacks.
+     */
+    struct ContextMenuData {
+        SidebarToolbar* toolbar;
+        SidebarActions actions;
+    };
+
+    /**
+     * The signals connected to the context menu items. This must be kept track
+     * of so the data can be deallocated safely.
+     */
+    std::vector<std::tuple<GtkWidget*, gulong, std::unique_ptr<ContextMenuData>>> contextMenuSignals;
```

Comment:
See earlier comment: I think it would be nice to factor this out into a `SidebarPreviewLayersContextMenu` class (or similar) that both instances of `SidebarPreviewLayers` have a pointer to.